<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Fold changes • foldchanger</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Fold changes">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">foldchanger</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/detection_thresholds.html">Detection Thresholds</a></li>
    <li><a class="dropdown-item" href="../articles/dosage_calculations.html">Fold changes</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Fold changes</h1>
            
      

      <div class="d-none name"><code>dosage_calculations.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Detection dosage changes of genes within cancer cells can have
important implications for prognosis and treatment. Cancer samples
comprise of a mixture of cells from the cancer itself and normal patient
tissue (stroma).</p>
<p>Consequently, DNA extracted from cancer samples is a mixture of DNA
from cancer cells and stromal cells.</p>
<p>Using Illumina next generation sequencing (NGS), gene dosage can be
calculated using “depth-of-coverage” methods: the number of sequencing
reads mapping to a particular locus is used to infer the dosage of that
locus within the sample.</p>
<p>The <strong>observed</strong> number of reads can be compared to the
<strong>expected</strong> number of reads for that gene locus. The
expected value is derived from a cohort of normal tissue samples tested
with the same NGS process.</p>
</div>
<div class="section level2">
<h2 id="equations">Equations<a class="anchor" aria-label="anchor" href="#equations"></a>
</h2>
<p>There are different ways of presenting the relationship between
observed and expected read depth.</p>
<p>Firstly, you can take the log2 ratio of observed vs expected read
depth.</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>l</mi><mi>o</mi><mi>g</mi><mn>2</mn><mi>r</mi><mi>a</mi><mi>t</mi><mi>i</mi><mi>o</mi><mo>=</mo><mi>l</mi><mi>o</mi><mi>g</mi><mn>2</mn><mrow><mo stretchy="true" form="prefix">(</mo><mi>o</mi><mi>b</mi><mi>s</mi><mi>e</mi><mi>r</mi><mi>v</mi><mi>e</mi><mi>d</mi><mi>/</mi><mi>e</mi><mi>x</mi><mi>p</mi><mi>e</mi><mi>c</mi><mi>t</mi><mi>e</mi><mi>d</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">
log2ratio = log2(observed / expected)
</annotation></semantics></math></p>
<p>Alternatively, a “fold change” can be calculated. Two separate fold
change calculations are used by Qiagen for calculating gains and losses
of genetic information (described in the CLC handbook, section <a href="https://resources.qiagenbioinformatics.com/manuals/clcgenomicsworkbench/2305/index.php?manual=Copy_number_fold_change.html" class="external-link">Copy
number and fold change</a>).</p>
<p>Fold change for gains is calculated as:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>f</mi><mi>c</mi><mi>g</mi><mi>a</mi><mi>i</mi><mi>n</mi><mo>=</mo><mi>o</mi><mi>b</mi><mi>s</mi><mi>e</mi><mi>r</mi><mi>v</mi><mi>e</mi><mi>d</mi><mi>/</mi><mi>e</mi><mi>x</mi><mi>p</mi><mi>e</mi><mi>c</mi><mi>t</mi><mi>e</mi><mi>d</mi></mrow><annotation encoding="application/x-tex">
fcgain = observed / expected
</annotation></semantics></math></p>
<p>Fold change for losses is calculated as:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>f</mi><mi>c</mi><mi>l</mi><mi>o</mi><mi>s</mi><mi>s</mi><mo>=</mo><mo>−</mo><mrow><mo stretchy="true" form="prefix">(</mo><mi>e</mi><mi>x</mi><mi>p</mi><mi>e</mi><mi>c</mi><mi>t</mi><mi>e</mi><mi>d</mi><mi>/</mi><mi>o</mi><mi>b</mi><mi>s</mi><mi>e</mi><mi>r</mi><mi>v</mi><mi>e</mi><mi>d</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">
fcloss = - (expected / observed)
</annotation></semantics></math></p>
</div>
<div class="section level2">
<h2 id="rationale">Rationale<a class="anchor" aria-label="anchor" href="#rationale"></a>
</h2>
<p>Having two different equations for fold change may seem confusing,
but the reason is to allow greater precision in dosage measurement.</p>
<p>The fold change loss equation provides more differentiation between
coverage levels when fewer reads are observed than expected.</p>
<p><img src="dosage_calculations_files/figure-html/plot1-1.png" width="700"></p>
<p>Conversely, the fold change gain equation is more useful when gene
amplifications are present: the number of observed reads may be several
fold higher than expected.</p>
<p><img src="dosage_calculations_files/figure-html/plot2-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="simulated-data">Simulated data<a class="anchor" aria-label="anchor" href="#simulated-data"></a>
</h2>
<p>We can see how the different equations present an artificial dataset.
This fake chromosome has a region of normal dosage, then an amplified
region, then a region with a dosage loss.</p>
<p>The log2 ratio looks like this.</p>
<p><img src="dosage_calculations_files/figure-html/log2r-plot-1.png" width="700"></p>
<p>The fold change gain equation looks like this. Notice how the
amplified region is “pulled up” and is more separated from the normal
dosage region, and that the normal dosage region now has an average
value of 1, not 0.</p>
<p><img src="dosage_calculations_files/figure-html/fcgain-plot-1.png" width="700"></p>
<p>And the fold change loss equation looks like this. In the reverse to
the gain equation, the loss equation pulls the deleted region further
from the normal dosage baseline, and the normal dosage region now has an
average value of -1.</p>
<p><img src="dosage_calculations_files/figure-html/fcloss-plot-1.png" width="700"></p>
<p>We can also combine the fold change equations, so that the gain
equation is used when the observed dosage is higher than expected, and
the loss equation is used when the observed dosage is lower than
expected. This results in a “splitting” of the central cluster of normal
dosage points.</p>
<p><img src="dosage_calculations_files/figure-html/fccombined-plot-1.png" width="700"></p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Joe Shaw.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
